airflow/www/utils.py

```diff
@@ -68,6 +71,33 @@ def limit_sql(sql, limit, conn_type):
             """.format(**locals())
     return sql
 
+
+def action_logging(action):
+    '''
+    Decorator to log user actions
+    '''
+    def decorator(f):
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            session = settings.Session()
+
+            if hasattr(login.current_user, 'username'):
+                user = login.current_user.username
+            else:
+                user = 'anonymous'
+
+            session.add(
+                models.Log(
+                    event=action,
+                    task_instance=None,
+                    owner=user
```

Comment:
it'd be really nice to also store a string representation of args and kwargs as well. Maybe we can add an `extra` blob to the Log model and store a json representation of both these things as in `{"args": ['foo'], "kwargs": {"key": "bar"}}`

We'll need a db migration script to add the field. It's easy but unintuitive at first. I can guide you through the process.